### ISSUE TITLE
[BUG] Doc: what is a role?

### ISSUE THREAD

**Issue by daks** [saltstack-formulas/template-formula#248]
### Formula commit hash / release tag
master

### Versions reports (master & minion)
not relevant

### Pillar / config used
not relevant

---

### Describe the bug

I was reading doc to better understand map.jinja/TOFS override for formulas. The `docs/map.jinja.rst` talks about "role" but without explaining what it is and how it works. I think it make this part of the doc cryptic without some previous knowledge.

**Comment by baby-gnu** [comment]
I agree that `role` is not straightforward since it's a pure user convention.

I'll propose a little introduction paragraph to explain the idea behind the `roles`.
